dsc/workflows/base/__init__.py

```diff
@@ -21,51 +24,73 @@
-        s3_bucket: str,
-        batch_id: str,
         collection_handle: str,
-        output_queue: str,
+        batch_id: str,
     ) -> None:
         """Initialize base instance.
 
         Args:
-            workflow_name: The name of the workflow.
-            submission_system: The system to which item submissions will be sent
-                (e.g. DSpace@MIT).
-            email_recipients: The email addresses to notify after runs of
-                the workflow.
-            metadata_mapping: A mapping file for generating DSpace metadata
-                from the workflow's source metadata.
-            s3_bucket: The S3 bucket containing bitstream and metadata files for
-                the workflow.
+            collection_handle: The handle of the DSpace collection to which
+                submissions will be uploaded.
             batch_id: Unique identifier for a 'batch' deposit that corresponds
                 to the name of a subfolder in the workflow directory of the S3 bucket.
                 This subfolder is where the S3 client will search for bitstream
                 and metadata files.
-            collection_handle: The handle of the DSpace collection to which
-                submissions will be uploaded.
-            output_queue: The SQS output queue used for retrieving result messages
-                from the workflow's submissions.
         """
-        self.workflow_name: str = workflow_name
-        self.submission_system: str = submission_system
-        self.email_recipients: list[str] = email_recipients
-        self.metadata_mapping: dict = metadata_mapping
-        self.s3_bucket: str = s3_bucket
-        self.batch_id: str = batch_id
-        self.collection_handle: str = collection_handle
-        self.output_queue: str = output_queue
+        self.batch_id = batch_id
+        self.collection_handle = collection_handle
 
     @property
     def batch_path(self) -> str:
         return f"{self.workflow_name}/{self.batch_id}"
 
+    @property
+    def metadata_mapping(self) -> dict:
+        with open(self.metadata_mapping_path) as mapping_file:
+            return json.load(mapping_file)
+
+    @final
+    @classmethod
+    def load(
+        cls, workflow_name: str, collection_handle: str, batch_id: str
+    ) -> BaseWorkflow:
+        """Return configured workflow class instance.
+
+        Args:
+            workflow_name: The label of the workflow. Must match a key from
+            config.WORKFLOWS.
+            collection_handle: The handle of the DSpace collection to which the batch will
+            be submitted.
+            batch_id: The S3 prefix for the batch of DSpace submissions.
+        """
+        workflow_class = cls.get_workflow(workflow_name)
+        return workflow_class(
+            collection_handle=collection_handle,
+            batch_id=batch_id,
+        )
+
+    @final
+    @classmethod
+    def get_workflow(cls, workflow_name: str) -> type[BaseWorkflow]:
```

Comment:
I'm curious if you considered the `__subclasses()__` approach for discovering workflow classes?  

I'm not opposed to the dictionary/JSON object configuration approach either, it's certainly worked in other projects (e.g. Transmog).  But in those situations, there was often more configuration about the thing in that config object.  As-is, I think there is a bit of a tension between the class level attributes and this config object.  How do you decide what goes where?  

If the config object is _only_ for discovering the python classes based on a name, I do think the `__subclasses()__` approach might be worth considering.

Reply:
Found a much cleaner approach with `__subclasses__`, great suggestion!